### PR TITLE
Change endpoint in meshconfig(platform: Docker)

### PR DESCRIPTION
**Pull request by metonymic-smokey**
**Description**
This PR changes the endpoint in the meshconfig when `platform: Docker` to reflect the port specified by the user(PORT) in meshconfig's endpoint. Changes the endpoint to `http://localhost:PORT` based on the user's confirmation.
Abstracts this into a function in `platform.go` to change based on the platform.

**Screenshots**
1. Before start:
image

2. After start:
image

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by navendu-pottekkat — COMMENTED** [review]
Does this mean that a user can only configure the port to which Meshery is deployed to?

**Comment by metonymic-smokey** [comment]
Yes, that's how it was earlier as well(for docker).

**Comment by metonymic-smokey** [comment]
@leecalcote could you review this pls?
Thanks!

**Comment by leecalcote** [comment]
Yes, I'm having a look now... Docker as a platform is a special case as we think of it today. If we hardcode `localhost`, we're assuming that all Docker-based deployments are to be done locally, which for now, is ok, and time will tell if any user ever has the need for remote administration of a Docker-based, Meshery deployment.

**Comment by leecalcote** [comment]
While our use case for Docker-based deployments is local-only, I think we want to avoid rewriting the hostname/IP address in the endpoint URL. As a user, I know that such an update would come as a surprise to me. If we really consider that the endpoint the user has listed should be updated, what do you think of prompting the user to confirm the change? Doing so hopefully has two positive effects:

1. Meshery gets to show off how intelligent it is to the user.
2. The user appreciates Meshery's intelligence and remains comfortable in the thought they are in control; that Meshery is there to help them, offering helpful suggestions. Trust is built with the user.

**Comment by metonymic-smokey** [comment]
@leecalcote yes, that's a good point with regard to the user experience.   
To that end, how about using the `-y` flag and asking for a confirmation before changing the endpoint to localhost? That way, a user has a choice to deny the change.

**Comment by leecalcote** [comment]
> @leecalcote yes, that's a good point with regard to the user experience.
> To that end, how about using the `-y` flag and asking for a confirmation before changing the endpoint to localhost? That way, a user has a choice to deny the change and this can be automated too, if required. It will also ensure a uniform UX across commands if we use the same flag.

This sounds spot on. 👍

File: mesheryctl/internal/cli/root/system/start.go [hunk 1]
```diff
@@ -193,7 +195,27 @@ func start() error {
...
+		userResponse := false
+		if utils.SilentFlag {
+			userResponse = true
+		} else {
+			// ask user for confirmation
+			userResponse = utils.AskForConfirmation("The endpoint address will be changed to localhost. Are you sure you want to continue?")
+		}
+
+		if userResponse {
+			endpoint.Address = utils.EndpointProtocol + "://localhost"
```
Comment (kumarabd) [inline mesheryctl/internal/cli/root/system/start.go:208]: @metonymic-smokey does this mean that the endpoint is localhost always in a docker deployment?

Comment (metonymic-smokey) [reply mesheryctl/internal/cli/root/system/start.go:208]: @kumarabd no, not always.
It is changed to `localhost` based on the user's confirmation.

**Review by kumarabd — APPROVED** [review]
LGTM

**Comment by leecalcote** [comment]
Just a couple failing lint checks to be addressed here...

**Comment by kumarabd** [comment]
yes I didn't realise this, ill fix this one.

**Comment by metonymic-smokey** [comment]
> Just a couple failing lint checks to be addressed here...

@leecalcote I had run `golangci-lint` locally and it did not throw up any lint errors.
I remember the lint check failed because some package was missing...

**Comment by metonymic-smokey** [comment]
> Just a couple failing lint checks to be addressed here...

@leecalcote the lint checks passed for my commits till yesterday...

**Comment by leecalcote** [comment]
No sweat, @metonymic-smokey. They were a very simple `gofmt -s`  check. This note was in part for Abishek to consider as the PR is/was merged.

**Pull request merged**